docs/recipes/solid.md

```diff
@@ -0,0 +1,178 @@
+# Usage with SolidJS
+
+XState can be used with SolidJS to:
+
+- Coordinate local state
+- Manage global state
+- Consume data from other hooks
+- etc.
+
+## Local state
+
+```js
+import { useMachine } from '@xstate/solid';
+import { toggleMachine } from '../path/to/toggleMachine';
+
+function Toggle() {
+  const [current, send] = useMachine(toggleMachine);
+
+  return (
+    <button onClick={() => send('TOGGLE')}>
+      {current.matches('inactive') ? 'Off' : 'On'}
+    </button>
+  );
+}
+```
+
+## Global State in Context
+
+### Context Provider
+You can store a machine or service in [SolidJS context](https://www.solidjs.com/docs/latest/api#createcontext) to make them available throughout the component tree. Because @xstate/solid produces machines, actors, and services with SolidJS' fine-grained reactivity, you can access machine context and methods (e.g. `matches` or `can`) without worrying about wasted re-renders
+```js
+import { createContext } from 'solid-js';
+import { useMachine } from '@xstate/solid';
+import { authMachine } from './auth.machine';
+
+export const GlobalStateContext = createContext({});
+
+export const GlobalStateProvider = (props) => {
+  const authService = useMachine(authMachine);
+
+  return (
+    <GlobalStateContext.Provider value={{ authService }}>
+      {props.children}
+    </GlobalStateContext.Provider>
+  );
+};
+```
+
+> For Typescript, you can create the context as `createContext({ authService: {} as UseMachineReturn<typeof authMachine> });` to ensure strong typings.
+
+### Utilizing context
+
+Further down the tree, you can subscribe to the service like this:
+
+```js
+import { useContext } from 'solid-js';
+import { GlobalStateContext } from './globalState';
+
+export const SomeComponent = (props) => {
+  const {authService: [state]} = useContext(GlobalStateContext);
```

Comment:
note that it looks like perhaps Prettier was not run over those files, it would be great to reformat all changes automatically to avoid unrelated, accidental, changes in some future PRs

Reply:
Weird this wasn't formatted, I've been running the git hooks before every commit. I tried running prettier manually over the docs and several other docs not related to this PR were changed.